src/drag_and_drop.rs

```diff
@@ -0,0 +1,73 @@
+//! Simple drag and drop example
+//!
+//! Ported over from example code:
+//! https://developer.gnome.org/gtkmm-tutorial/stable/sec-dnd-example.html.en
+extern crate gdk;
+extern crate gtk;
+
+use gtk::prelude::*;
+
+fn main() {
+    if gtk::init().is_err() {
+        println!("Failed to initialize GTK.");
+        return;
+    }
+
+    // Configure button as drag source
+    let button = gtk::Button::new_with_label("Drag here");
+    let targets = vec![
+        gtk::TargetEntry::new("STRING", gtk::TargetFlags::empty(), 0),
+        gtk::TargetEntry::new("text/plain", gtk::TargetFlags::empty(), 0),
+    ];
+    button.drag_source_set(gdk::MODIFIER_MASK,
+                           &targets,
+                           gdk::ACTION_COPY);
+    button.connect_drag_data_get(|_, _, s, _, _| {
+        println!("drag-data-get");
+        let data = "I'm data!";
+        s.set_text(data, data.len() as i32);
+    });
+    button.connect_drag_begin(|_, c| {
+        println!("drag-begin");
+        for t in c.list_targets() {
+            println!("{}", t.name());
+        }
+    });
+    button.connect_drag_end(|_, _| {
+        println!("drag-end");
+    });
+
+    // Configure label as drag destination
+    let label = gtk::Label::new("Drop here");
+    label.drag_dest_set(gtk::DEST_DEFAULT_ALL,
```

Comment:
If set gtk::DEST_DEFAULT_MOTION | gtk::DEST_DEFAULT_HIGHLIGHT and 
```
    label.connect_drag_drop(move |w, c, _, _, time| {
        println!("drag-drop");
        w.drag_get_data(c, &atom, time);
        true
    });
```
then `drag-data-received` called. May it be fixed with `Inhibit`